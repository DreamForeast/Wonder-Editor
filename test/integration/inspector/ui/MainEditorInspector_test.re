open Wonder_jest;

open Expect;

open Expect.Operators;

open Sinon;

let _ =
  describe("MainEditorInspector", () => {
    let sandbox = getSandboxDefaultVal();
    beforeEach(() => {
      sandbox := createSandbox();
      MainEditorSceneTool.initStateAndGl(~sandbox, ());
      MainEditorSceneTool.createDefaultScene(
        sandbox,
        MainEditorAssetTool.initAssetTree,
      );
      EventListenerTool.buildFakeDom()
      |> EventListenerTool.stubGetElementByIdReturnFakeDom;
    });
    afterEach(() => restoreSandbox(refJsObjToSandbox(sandbox^)));
    describe("change source to show it's inspector", () => {
      test("if not set currentSelectSource, show nothing", () =>
        BuildComponentTool.buildInspectorComponent(
          TestTool.buildEmptyAppState(),
          InspectorTool.buildFakeAllShowComponentConfig(),
        )
        |> ReactTestTool.createSnapshotAndMatch
      );
      describe("else set currentSelectSource is SceneTree", () => {
        beforeEach(() => {
          MainEditorSceneTool.setFirstBoxTobeCurrentSceneTreeNode();

          CurrentSelectSourceEditorService.setCurrentSelectSource(
            EditorType.SceneTree,
          )
          |> StateLogicService.getAndSetEditorState;
        });
        test("show currentSceneTreeNode component", () =>
          BuildComponentTool.buildInspectorComponent(
            TestTool.buildEmptyAppState(),
            InspectorTool.buildFakeAllShowComponentConfig(),
          )
          |> ReactTestTool.createSnapshotAndMatch
        );
      });
      describe("else set currentSelectSource is Asset", () => {
        beforeEach(() =>
          CurrentSelectSourceEditorService.setCurrentSelectSource(
            EditorType.Asset,
          )
          |> StateLogicService.getAndSetEditorState
        );
        test("show currentNodeId's asset node component", () => {
          let assetTreeDomRecord =
            MainEditorAssetTool.buildTwoLayerAssetTreeRoot();
          let component = BuildComponentTool.buildAssetComponent();

          assetTreeDomRecord
          |> MainEditorAssetNodeTool.OperateTwoLayer.getFirstFolderDomIndexForAssetTree
          |> MainEditorAssetTool.clickAssetTreeNodeToSetCurrentNode(component);

          BuildComponentTool.buildInspectorComponent(
            TestTool.buildEmptyAppState(),
            InspectorTool.buildFakeAllShowComponentConfig(),
          )
          |> ReactTestTool.createSnapshotAndMatch;
        });
      });
    });
  });