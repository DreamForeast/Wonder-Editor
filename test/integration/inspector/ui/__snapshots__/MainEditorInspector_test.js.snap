// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MainEditorInspector change source to show it's inspector else set currentSelectSource is Asset show currentNodeId's asset node component 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-assetTree"
  >
    <div
      className=""
    >
      <h1>
        Folder
      </h1>
      <hr />
      <span
        className=""
      >
        name:
      </span>
      <input
        className="input-component float-input"
        disabled={false}
        onBlur={[Function]}
        onChange={[Function]}
        type="text"
        value="newFolder"
      />
    </div>
  </article>
</article>
`;

exports[`MainEditorInspector change source to show it's inspector else set currentSelectSource is SceneTree show currentSceneTreeNode component 1`] = `
<article
  className="wonder-inspector-component"
>
  <article
    className="wonder-inspector-sceneTree"
  >
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          Name
        </div>
      </div>
      <div>
        <article
          className="wonder-string-input"
        >
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="cube"
          />
        </article>
      </div>
    </article>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          transform
        </div>
      </div>
      <article
        className="wonder-inspector-transform"
      >
        <article
          className="wonder-float-input"
        >
          <span
            className="component-label"
          >
            X : 
          </span>
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </article>
        <article
          className="wonder-float-input"
        >
          <span
            className="component-label"
          >
            Y : 
          </span>
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </article>
        <article
          className="wonder-float-input"
        >
          <span
            className="component-label"
          >
            Z : 
          </span>
          <input
            className="input-component float-input"
            onBlur={[Function]}
            onChange={[Function]}
            type="text"
            value="0"
          />
        </article>
      </article>
    </article>
    <article
      className="componentBox-component"
    >
      <div
        className="header"
      >
        <div
          className="header-triangle"
          onClick={[Function]}
        >
          <span
            className="triangle-bottom"
          />
        </div>
        <div
          className="header-title"
        >
          basicMaterial
        </div>
      </div>
      <article
        className="wonder-inspector-material"
      >
        <article
          className="wonder-material-color"
        >
          <div
            className=""
          >
            <span
              className=""
            >
              color : 
            </span>
            <span
              className=""
            >
              #ffffff
            </span>
            <button
              className=""
              onClick={[Function]}
            >
              pick color
            </button>
          </div>
        </article>
        <article
          className="wonder-material-texture"
        >
          <div
            className="texture_ground"
            onDragEnter={[Function]}
            onDragLeave={[Function]}
            onDragOver={[Function]}
            onDrop={[Function]}
            style={
              Object {
                "opacity": "1",
              }
            }
          />
          <span
            className=""
          >
            texture:
          </span>
          <img
            src="./public/img/null.jpg"
          />
          <button
            className="texture_remove"
            onClick={[Function]}
          >
            remove
          </button>
        </article>
      </article>
    </article>
    <article
      className="wonder-addable-component"
    >
      <button
        disabled={false}
        onClick={[Function]}
      >
        add component
      </button>
    </article>
  </article>
</article>
`;

exports[`MainEditorInspector change source to show it's inspector if not set currentSelectSource, show nothing 1`] = `
<article
  className="wonder-inspector-component"
/>
`;
